packages/compass-query-bar/src/components/query-bar/query-bar.tsx

```diff
@@ -49,74 +60,218 @@ const openQueryHistoryStyles = cx(
   focusRingStyles
 );
 
+const rowStyles = css({
+  display: 'flex',
+  flexGrow: 1,
+});
+
 type QueryBarProps = {
+  autoPopulated: boolean;
   buttonLabel?: string;
   expanded: boolean;
-  isQueryOptionsExpanded?: boolean;
-  valid: boolean;
+  filterValid: boolean;
+  layout?: QueryBarLayout;
+  onApply: () => void;
+  onChangeQueryOption: (queryOption: QueryOption, value: string) => void;
+  onReset: () => void;
   queryState: 'apply' | 'reset';
+  refreshEditorAction: () => void;
+  schemaFields: string[];
+  serverVersion: string;
   showQueryHistoryButton?: boolean;
   toggleExpandQueryOptions: () => void;
   toggleQueryHistory: () => void;
+  valid: boolean;
 };
 
 export const QueryBar: React.FunctionComponent<QueryBarProps> = ({
+  autoPopulated,
   buttonLabel = 'Apply',
   expanded: isQueryOptionsExpanded = false,
-  valid: isQueryValid,
+  filterValid: isFilterValid,
+  layout = [
+    'filter',
+    'project',
+    ['sort', 'maxTimeMS'],
+    ['collation', 'skip', 'limit'],
+  ],
+  onApply: _onApply,
+  onChangeQueryOption,
+  onReset: _onReset,
   queryState,
+  refreshEditorAction,
+  schemaFields,
+  serverVersion,
   showQueryHistoryButton = true,
   toggleExpandQueryOptions,
-  toggleQueryHistory,
+  toggleQueryHistory: _toggleQueryHistory,
+  valid: isQueryValid,
+
+  // This form uses extra props from the store for placeholders and
+  // input values for the query options. `filterValue`, `sortValue`, etc.
+  ...props
 }) => {
+  const onReset = useCallback(
+    (evt: React.MouseEvent) => {
+      // Prevent form submission.
+      evt.preventDefault();
+
+      _onReset();
+    },
+    [_onReset]
+  );
+
+  const toggleQueryHistory = useCallback(
+    (evt: React.MouseEvent) => {
+      // Prevent form submission.
+      evt.preventDefault();
+
+      _toggleQueryHistory();
+    },
+    [_toggleQueryHistory]
+  );
+
+  const onApply = useCallback(() => {
+    if (isQueryValid) {
+      _onApply();
+    }
+  }, [_onApply, isQueryValid]);
+
+  const renderQueryOption = useCallback(
```

Comment:
In the follow up pr we'll make this all part of the `QueryOption` component interface so we won't need this function declaration happening here.